include/geode/mesh/core/detail/facet_storage.h

```diff
@@ -87,21 +110,21 @@ namespace geode
                 const auto itr = facet_indices_.find( vertices );
                 if( itr != facet_indices_.end() )
                 {
-                    return facet_indices_.at( vertices );
+                    return itr->second.id;
                 }
                 return NO_ID;
             }
 
             index_t add_facet( const VertexCycle& vertices )
             {
-                const auto id = find_facet( vertices );
+                auto& id = facet_indices_[vertices].id;
                 if( id != NO_ID )
                 {
                     counter_->set_value( id, counter_->value( id ) + 1 );
                     return id;
                 }
                 const auto size = facet_indices_.size();
-                facet_indices_[vertices] = size;
+                id = size;
                 facet_attribute_manager_.resize( size + 1 );
                 vertices_->modify_value(
                     size, SetValue{ vertices.vertices() } );
```

Comment:
```suggestion
                    id, SetValue{ vertices.vertices() } );
```
Same thing in return